src/IntegrationTests.jl

```diff
@@ -1,5 +1,159 @@
 module IntegrationTests
 
-# Write your package code here.
+using Pkg
+using PkgDependency
+
+export depending_projects
+
+"""
+    depending_projects(
+        package_name::AbstractString, 
+        package_filter::Union{TString,Regex} where {TString<:AbstractString}, 
+        project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project())
+    ) -> Vector{String}
+
+Returns a list of packages that have the package `package_name` as a dependency. 
+
+# Arguments
+
+- `package_name`: Name of the dependency
+- `package_filter`: Ignore all packages that do not match `package_filter`. This includes the 
+        top node package of the graph. Child nodes always are checked for `package_name`, but 
+        they are not traveres if they do not match `package_filter`.
+- `project_tree`: Project tree, where to search the dependent packages. Each (sub-) package 
+        needs to be AbstractDict{String, AbstractDict}
+
+# Returns
+
+all packages which have the searched dependency
+
+"""
+function depending_projects(
+    package_name::AbstractString,
+    package_filter::Union{TString,Regex},
+    project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project()),
+)::Vector{String} where {TString<:AbstractString}
+    packages::Vector{String} = []
+    visited_packages::Vector{String} = []
+    _traverse_tree!(package_name, package_filter, project_tree, packages, visited_packages)
+    return packages
+end
+
+"""
+    depending_projects(
+        package_name::AbstractString, 
+        package_filter::AbstractVector{TString} where {TString<:AbstractString}, 
+        project_tree::AbstractDict=PkgDependency.builddict(Pkg.project().uuid, Pkg.project())
+    ) -> Vector{String}
+
+Returns a list of packages that have the package `package_name` as a dependency. 
+
+# Arguments
+    
+- `package_name`: Name of the dependency
+- `package_filter`: Ignore all packages that do not match `package_filter`. This includes the 
+        top node package of the graph. Child nodes always are checked for `package_name`, but 
+        they are not traveres if they do not match `package_filter`. Must be not empty.
+- `project_tree`: Project tree, where to search the dependent packages. Each (sub-) package 
+        needs to be AbstractDict{String, AbstractDict}
+    
+!!! note 
+    If you want to use a combination of srings and regexes in a vector, you should use a pure 
+    regex instead.
+
+# Returns
+
+all packages which have the searched dependency
+
+"""
```

Comment:
Is this duplication of the documentation necessary? It should just show a single one for both of the functions if it's missing, and almost all of this is the same as the other version of the function.